src/tools/perf/ucp_tests.cc

```diff
@@ -98,8 +107,8 @@ class ucp_perf_test_runner {
 
         switch (CMD) {
         case UCX_PERF_CMD_TAG:
-            if (UCP_PERF_DATATYPE_IOV == m_perf.params.ucp.datatype) {
-                buffer   = m_perf.ucp.iov;
+            if (UCP_PERF_DATATYPE_IOV == m_perf.params.ucp.send_datatype) {
```

Comment:
can we prepare "datatype" up front and keep it in a member variable? to avoid that if on runtime. same for recv.